tests/kafkatest/tests/client/consumer_test.py

```diff
@@ -612,3 +612,73 @@ def test_valid_assignment(self, assignment_strategy=None, metadata_quorum=quorum
                 err_msg="expected valid assignments of %d partitions when num_started %d: %s" % \
                         (self.NUM_PARTITIONS, num_started, \
                          [(str(node.account), a) for node, a in consumer.current_assignment().items()]))
+
+class ConsumerProtocolMigrationTest(VerifiableConsumerTest):
+    TOPIC = "test_topic"
+    NUM_PARTITIONS = 6
+
+    RANGE = "org.apache.kafka.clients.consumer.RangeAssignor"
+    ROUND_ROBIN = "org.apache.kafka.clients.consumer.RoundRobinAssignor"
+    STICKY = "org.apache.kafka.clients.consumer.StickyAssignor"
+    COOPERATIVE_STICKEY = "org.apache.kafka.clients.consumer.CooperativeStickyAssignor"
+    all_assignment_strategies = [RANGE, ROUND_ROBIN, COOPERATIVE_STICKEY, STICKY]
+
+    def __init__(self, test_context):
+        super(ConsumerProtocolMigrationTest, self).__init__(test_context, num_consumers=2, num_producers=1,
+                                                            num_zk=0, num_brokers=2, topics={
+                self.TOPIC : { 'partitions': self.NUM_PARTITIONS, 'replication-factor': 1 }
+            })
+
+    @cluster(num_nodes=6)
+    @matrix(
+        clean_shutdown=[True, False],
+        enable_autocommit=[True, False],
+        metadata_quorum=[quorum.isolated_kraft],
+        use_new_coordinator=[True],
+        consumer_group_migration_policy=["bidirectional"],
+        assignment_strategy=all_assignment_strategies,
+        bounce_mode=["all", "rolling"]
+    )
+    def test_boucing_consumer(self, clean_shutdown, enable_autocommit, metadata_quorum, use_new_coordinator,
+                                      consumer_group_migration_policy, assignment_strategy, bounce_mode):
+        producer = self.setup_producer(self.TOPIC)
+        consumer = self.setup_consumer(self.TOPIC, group_protocol=consumer_group.consumer_group_protocol,
+                                       assignment_strategy=assignment_strategy, enable_autocommit=enable_autocommit)
+        self.mark_for_collect(consumer, 'verifiable_consumer_stdout')
+        consumer.start()
+
+        producer.start()
+        self.await_produced_messages(producer)
+
+        consumer.start()
+        self.await_all_members(consumer)
+        self.await_consumed_messages(consumer)
+
+        # consumer.group_protocol = consumer_group.consumer_group_protocol
```

Comment:
Don’t you need this to upgrade to the new protocol?

Reply:
It's temporarily commented out for debug. The errors remain the same even if the protocol is not upgraded. We'll need to change it back after finding the root cause